### DISCUSSION TITLE
Linking to URLs after matching regex

### DISCUSSION THREAD

**Discussion by TheLocehiliosan** [ghostty-org/ghostty#3754] [Feature Requests, Ideas]
iTerm has a feature called "Smart selections" which I use extensively to do things like make JIRA tickets numbers links that open the ticket or change linter errors codes to links that open the documentation. I've been trying to find a way to do the same with Ghostty, but I'm lost. Is there a way to do it? It seems like the "link" option exists but is not yet supported.

A simple use-case I'm trying to implement would be to have any text like `SC2236` become a link like `https://www.shellcheck.net/wiki/SC2236`. With iTerm, I match the regex `SC\d+` and then map it to a URL like `https://www.shellcheck.net/wiki/\0`.

Is this possible with Ghostty?

**Comment by nklaassen** [comment]
Yeah I also use this feature in iTerm2 to make github PR/issue numbers (like `#12345`) clickable links to that PR/issue in a specific github repo, it's the number 1 thing i miss in ghostty so far

**Comment by joshbode** [comment]
This sounds related to https://github.com/ghostty-org/ghostty/discussions/4379 - I've got a branch linked there which adds support for the `link` option which I'll submit as a PR if there's interest.

The only challenge with your use-case is that the `link` points to the default system opener (e.g. `open` or `xdg-open`), like the default URL matcher, so you'd probably need to make a wrapper for the system opener to augment the matches (e.g. to add the base URL).

I have an opener wrapper that I use to redirect certain matches to a Neovim session which I was using with Alacritty and can use with Ghostty (with the changes I made). I can provide some details of the wrapper, if that would help :)

**Comment by nklaassen** [comment]
iTerm just lets you use a regular expression to make a hyperlink, which is pretty convenient. You can substitute regex matches into the url.
<img width="720" alt="Screenshot 2025-01-03 at 5 24 54 PM" src="https://github.com/user-attachments/assets/d4a1a67e-3dc7-4728-98f5-302c3a1d7e2a" />

**Reply by danielhoherd** [reply]
This is exactly what I do. This is the one feature that is preventing me from adopting ghostty as my primary terminal emulator.

**Reply by pluiedev** [reply]
Please reread discussion rules and refrain from bumping older issues - you're sending everyone involved an email that does not meaningfully contribute to the current discussion.

**Reply by 3v1n0** [reply]
This has been there forever in tilix or gnome-terminal too

<img width="1088" height="203" alt="image" src="https://github.com/user-attachments/assets/38d07945-489a-4688-8fea-0060804dfcef" />
